static/src/javascripts/lib/element-inview.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+import bean from 'bean';
+import debounce from 'lodash/functions/debounce';
+import { noop } from 'lib/noop';
+
+type offsetType = {
+    left?: number,
+    top?: number,
+};
+
+const elementIsInView = (el: HTMLElement, offsets_?: offsetType): boolean => {
+    const offsets = Object.assign(
+        {},
+        { left: 0, right: 0, top: 0, bottom: 0 },
+        offsets_
+    );
+
+    const rect = el.getBoundingClientRect();
+    const viewportHeight = window.innerHeight;
+    const viewportWidth = window.innerWidth;
+
+    const fromTop = rect.top + offsets.top;
+    const fromBottom = rect.bottom - offsets.bottom;
+    const fromLeft = rect.left - offsets.left;
+    const fromRight = rect.right + offsets.right;
+
+    const visibleVertically = fromTop < viewportHeight && fromBottom > 0;
+    const visibleHorizontally = fromLeft < viewportWidth && fromRight > 0;
+
+    return visibleVertically && visibleHorizontally;
+};
+
+const elementInView = (
+    element: HTMLElement,
+    container: HTMLElement,
+    offsets?: offsetType
+): {
+    on: (eventName: 'firstview', callback: ?(el: HTMLElement) => void) => void,
+} => {
+    let hasBeenSeen = false;
+
+    const events = {
+        firstview: noop,
+    };
+
+    bean.on(
+        container,
+        'scroll',
+        debounce(() => {
+            const inView = elementIsInView(element, offsets);
```

Comment:
We're calling `elementIsInView` every time this function is called and assigning to `inView`, however we only check the value of `inView` if `events.firstview` is true. 

Could we optimise by adding the call to `elementIsInView` within an if condition which checks whether `elementIsInView` is true? That way we'd eliminate lot's of needless calls to `elementIsInView` (which call `getBoundingClientRect` which I believe forces a repaint.